R/initialisers.R

```diff
@@ -0,0 +1,108 @@
+#' initialise_analytics
+#'
+#' @param ga_code The Google Analytics code
+#' @importFrom magrittr %>%
+#' @return TRUE if written, FALSE if not
+#' @export
+#'
+#' @examples initialise_analytics(ga_code = "0123456789")
+initialise_analytics <- function(ga_code) {
+  is_valid_ga4_code <- function(ga_code) {
+    stringr::str_length(ga_code) == 10 & typeof(ga_code) == "character"
+  }
+
+  if (is_valid_ga4_code(ga_code) == FALSE) {
+    stop(
+      'You have entered an invalid GA4 code in the ga_code argument.
+      Please enter a 10 digit code as a character string.
+      e.g. "0123QWERTY"'
+    )
+  }
+
+  html_script <- "<script>
+// Define dataLayer and the gtag function.
+window.dataLayer = window.dataLayer || [];
+function gtag(){dataLayer.push(arguments);}
+
+// Default ad_storage to 'denied' as a placeholder
+// Determine actual values based on your own requirements
+gtag('consent', 'default', {
+  'ad_storage': 'denied',
+  'analytics_storage': 'denied'
+});
+</script>
+
+<!-- Global site tag (gtag.js) - Google Analytics -->
+  <script async src='https://www.googletagmanager.com/gtag/js?id=G-XXXXXXXXXX'></script>
+  <script>
+    window.dataLayer = window.dataLayer || [];
+    function gtag(){dataLayer.push(arguments);}
+
+    gtag('js', new Date());
+
+  gtag('config', 'G-XXXXXXXXXX');
+
+/*
+The custom trackers below can be tailored to match the inputs used in your dashboard.
+*/
+
+  $(document).on('change', 'select#selectPhase', function(e) {
+    gtag('event', 'select phase', {'event_category' : 'choose Phase',
+    'event_label' : document.querySelector('select#selectPhase').value
+    });
+  });
+
+  $(document).on('change', 'select#selectArea', function(e) {
+    gtag('event', 'select area', {'event_category' : 'choose Area',
+    'event_label' : document.querySelector('select#selectArea').value
+    });
+  });
+
+  $(document).on('click', 'ul#navlistPanel', function(e) {
+    gtag('event', 'navlistPanel', {'event_category' : 'navbar click',
+    'event_label' : document.querySelector('ul#navlistPanel > li.active > a').getAttribute('data-value')
+    });
+  });
+
+  $(document).on('click', 'ul#tabsetpanels', function(e) {
+    gtag('event', 'tab panels', {'event_category' : 'tab panel clicks',
+    'event_label' : document.querySelector('ul#tabsetpanels > li.active > a').getAttribute('data-value')
+    });
+  });
+
+
+  $(document).on('click', 'a#download_data', function(e) {
+    gtag('event', 'Download button', {'event_category' : 'Download button click'
+    });
+  });
+
+  $(document).on('shiny:disconnected', function(e) {
+    gtag('event', 'disconnect', {
+      'event_label' : 'Disconnect',
+      'event_category' : 'Disconnect'
+    });
+  });
+
+</script>
+      " %>% gsub("XXXXXXXXXX", ga_code, .)
```

Comment:
Is there a way we can rewrite to make the `.` explicit? It's flagging as a note in `devtools::check()`

Reply:
Changed to not using a pipe now